src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -524,18 +530,24 @@ def create_compile_context(self, target, target_workdir):
       )
     ]
 
+  @memoized_property
+  def _digested_rsc_native_image(self):
+    snapshot, = self.context._scheduler.capture_snapshots(tuple([
+      PathGlobsAndRoot(
+        PathGlobs(include=[self.get_options().rsc_native_image_location]),
+        root=get_buildroot(),
+      )
+    ]))
+    return snapshot
+
   def _runtool_hermetic(self, main, tool_name, args, distribution, tgt=None, input_files=tuple(), input_digest=None, output_dir=None):
     tool_classpath_abs = self.tool_classpath(tool_name)
     tool_classpath = fast_relpath_collection(tool_classpath_abs)
 
     # TODO(#6071): Our ExecuteProcessRequest expects a specific string type for arguments,
     # which py2 doesn't default to. This can be removed when we drop python 2.
-    str_jvm_options = [text_type(opt) for opt in self.get_options().jvm_options]
     cmd = [
-            distribution.java,
-          ] + str_jvm_options + [
-            '-cp', os.pathsep.join(tool_classpath),
-            main,
+            './{}'.format(self.get_options().rsc_native_image_location),
           ] + args
```

Comment:
It looks like in the zinc case you made use of the option conditional on whether it is set: I think that you'll need to do that here as well.